src/APL/env-apl.ts

```diff
@@ -0,0 +1,109 @@
+import { APL, AuthData } from "./apl";
+import { createAPLDebug } from "./apl-debug";
+
+const debug = createAPLDebug("EnvAPL");
+
+type AuthDataRequired = Omit<AuthData, "jwks" | "domain">;
+
+type Options = {
+  env: Record<keyof AuthDataRequired, string>;
+  /**
+   * Enable to log auth data to stdout.
+   * Do it once to save data in ENV and disable it later.
+   */
+  printAuthDataOnRegister?: boolean;
+};
+
+export class EnvApl implements APL {
+  private defaultOptions: Partial<Options> = {
+    printAuthDataOnRegister: false,
+  };
+
+  options: Options;
+
+  constructor(options: Options) {
+    if (!this.isAuthDataValid(options.env)) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        "EnvAPL constructor now filled with valid AuthData config. Try to install the app with \"printAuthDataOnRegister\" enabled and check console logs"
+      );
+    }
+
+    this.options = {
+      ...this.defaultOptions,
+      ...options,
+    };
+  }
+
+  private isAuthDataValid(authData: AuthData): boolean {
+    const keysToValidateAgainst: Array<keyof AuthData> = ["appId", "saleorApiUrl", "token"];
+
+    return keysToValidateAgainst.every(
+      (key) => authData[key] && typeof authData[key] === "string" && authData[key]!.length > 0
```

Comment:
Wonder if it will work with the vercel? 
https://nextjs.org/docs/basic-features/environment-variables#loading-environment-variables

Reply:
Can you elaborate? Vercel exposes env variables in `process.env.XYZ` but this APL doesn't even access it, everything is mapped in constructore

Reply:
You are right, constructor will take of this :bow:

Reply:
My concern was related to the way env vars are accessed, you have to do this the right way so it will be accessible by next-magic:
```
Note: In order to keep server-only secrets safe, environment variables are evaluated at build time, so only environment variables actually used will be included. This means that process.env is not a standard JavaScript object, so you’re not able to use [object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment). Environment variables must be referenced as e.g. process.env.PUBLISHABLE_KEY, not const { PUBLISHABLE_KEY } = process.env.
```

Reply:
Yes, thats why I decided not to read env inside APL, because it could get pruned by webpack. Mapping envs happen outside of SDK so it should be safe for Vercel/Next

Of course I will test it and attach it here